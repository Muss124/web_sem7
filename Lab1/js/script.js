function render(data) {
    var htmlWeather = document.getElementById("WeatherInfo")
    var htmlCity = document.getElementById("City")
    if (data['cod'] === 200) {
        //console.log("yes")
        htmlCity.textContent = data["name"]
        var template = "{{~it.weather :value:index}}<li>{{=value.parametr}} - {{=value.value}}</li>{{~}}"
        var json = {
            "weather": [
                { "parametr": "weather", "value": data["weather"][0]["description"] },
                { "parametr": "temperature (Celsius)", "value": data["main"]["temp"] },
                { "parametr": "humidity (%)", "value": data["main"]["humidity"] },
                { "parametr": "pressure (hPa)", "value": data["main"]["pressure"] },
                { "parametr": "wind speed (meter/sec)", "value": data["wind"]["speed"] }
            ]
        }
        var tempFn = doT.template(template)
        htmlWeather.innerHTML = tempFn(json)
    } else {
        //console.log("no")
        htmlCity.textContent = data["message"]
        htmlWeather.innerHTML = ""
    }
}

function getweather(city, callback) {
    var request = new XMLHttpRequest()
    request.open('GET', 'http://api.openweathermap.org/data/2.5/weather?q='.concat(city, "&APPID=f8e5ebb3f762d1a58aaff7f643d7410b&units=metric"), true)
    request.onload = function () {
        var data = JSON.parse(this.response)
        //console.log(data)
        callback(data)
    }
    request.send()
}

function formSubmit(event) {
    event.preventDefault()
    var city = document.getElementById("_citysearch").value;
    //console.log(city)
    getweather(city, render)    
}