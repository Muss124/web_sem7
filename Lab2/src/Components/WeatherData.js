import React from "react"
import store from "../store.js"

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		console.log("here");
		console.log(WeatherData)
		return (
			<div class="WeatherData">
				<div class="WeatherTop">
					<h2>{WeatherData["name"]}</h2>
					<img src={"http://openweathermap.org/img/wn/" + WeatherData["weather"][0]["icon"] + ".png"} alt=""></img>
					<h1>{WeatherData["main"]["temp"]} &#8451;</h1>
					<button type="submit" class="btn btn-secondary" onClick={() => {
						console.log("REMOVE");
						store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData["name"] });
					}} >&#10006;</button>
				</div>
				<div class="WeatherMain">
					<div class="WeatherMainEl">
						<div class="WeatherParam">Weather</div>
						<div class="WeatherValue">{WeatherData["weather"][0]["description"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Humidity (%)</div>
						<div class="WeatherValue">{WeatherData["main"]["humidity"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Pressure (hPa)</div>
						<div class="WeatherValue">{WeatherData["main"]["pressure"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Wind speed (meter/sec)</div>
						<div class="WeatherValue">{WeatherData["wind"]["speed"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Longitude </div>
						<div class="WeatherValue">{WeatherData["coord"]["lon"]}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Latitude</div>
						<div class="WeatherValue">{WeatherData["coord"]["lat"]}</div>
					</div>
				</div>
			</div>
		);
	}
}