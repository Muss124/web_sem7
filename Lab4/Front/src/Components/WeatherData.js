import React from "react"
import store from "../store.js"

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		console.log("here");
		console.log(WeatherData)
		return (
			<div class="WeatherData">
				<div class="WeatherTop">
					<h2>{WeatherData.CityName}</h2>
					<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
					<h1>{WeatherData.Temperature} &#8451;</h1>
					<button type="submit" class="btn btn-secondary" onClick={() => {
						console.log("REMOVE");
						store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData.CityName });
					}} >&#10006;</button>
				</div>
				<div class="WeatherMain">
					<div class="WeatherMainEl">
						<div class="WeatherParam">Weather</div>
						<div class="WeatherValue">{WeatherData.Weather}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Humidity (%)</div>
						<div class="WeatherValue">{WeatherData.Humidity}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Pressure (hPa)</div>
						<div class="WeatherValue">{WeatherData.Pressure}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Wind speed (meter/sec)</div>
						<div class="WeatherValue">{WeatherData.Wind}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Longitude </div>
						<div class="WeatherValue">{WeatherData.Longitude}</div>
					</div>
					<div class="WeatherMainEl">
						<div class="WeatherParam">Latitude</div>
						<div class="WeatherValue">{WeatherData.Latitude}</div>
					</div>
				</div>
			</div>
		);
	}
}