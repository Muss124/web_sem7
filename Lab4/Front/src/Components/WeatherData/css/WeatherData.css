.WeatherMain {
    margin-top   : auto;
    margin-bottom: auto;
}

.TopCityWeather .WeatherData .WeatherTop h1 {
    width     : 80%;
    text-align: right;
    font-size : 50px;
}

.WeatherTop {
    display: inline-block;
}

.WeatherMainEl {
    display         : flex;
    justify-content : space-between;
    background-color: #999;
    margin          : 5px;
    padding         : 5px;
}

.WeatherTop {
    display        : flex;
    justify-content: flex-start;
}


.WeatherParam {
    font-weight: bold;
}