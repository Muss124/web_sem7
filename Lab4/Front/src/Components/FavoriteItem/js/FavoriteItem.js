import React from 'react';
import {WeatherData} from "../../WeatherData/js/WeatherData"
import '../css/FavoriteItem.css'


export class FavoriteItem extends React.Component {
	render() {
		if (Object.keys(this.props.value).length === 1) {
			return (
				<div>
					{this.props.value.CityName}
					<div className="spinner-border text-danger" role="status">
						<span className="sr-only">Loading...</span>
					</div>
				</div>
			);
		}
		else {
			return (
				<div className="FavoriteItem">
					<WeatherData value={this.props.value} isMain={false}/>
				</div>
			);
		}
	}
}