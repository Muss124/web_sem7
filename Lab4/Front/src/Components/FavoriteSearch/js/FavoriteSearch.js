import React from "react"
import store from "../../../store"
import '../css/FavoriteSearch.css'
import FetchCityData from '../../../Actions/FetchCityData'
import RefreshCity from "../../../Actions/RefreshCity"

export class FavoriteSearch extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        this.addGeo(this.state.value);
        event.preventDefault();
    }

    addGeo(value) {
        console.log("form submited");
        store.dispatch({ type: "FAVORITE_ADD", payload: { CityName : value } });
        store.dispatch(FetchCityData(value));
    }
    render() {
        return (
            <div className="searchform">
                <div>
                    <h1>Favorite</h1>
                </div>
                <div className="searchformInside">
                    <form onSubmit={this.handleSubmit}>
                        <input type="search" value={this.state.value} onChange={this.handleChange}></input>
                        <button type="submit" className="btn btn-secondary btn-circle" >Add</button>
                    </form>
                    <button className="btn btn-secondary" onClick={() => {
                        console.log("REFRESH");
                        var refData = store.getState();
                        refData.forEach(el => {
                            console.log(el);
                            store.dispatch(RefreshCity(el.CityName));
                        });
                    }} >&#8634;</button>
                </div>
            </div>
        );
    }
}