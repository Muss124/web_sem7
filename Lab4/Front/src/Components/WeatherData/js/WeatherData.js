import React from "react"
import store from "../../../store"
import "../css/WeatherData.css"
import ServerDeleteCity from '../../../Actions/ServerDeleteCity'

export class WeatherData extends React.Component {
	render() {
		var WeatherData = this.props.value;
		//console.log("here");
		//console.log(WeatherData);
		//console.log(this.props.isMain);
		if (this.props.isMain) {
			return (
				<div className="WeatherData">
					<div className="WeatherTop">
						<h2>{WeatherData.CityName}</h2>
						<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
						<h1>{WeatherData.Temperature} &#8451;</h1>
					</div>
					<div className="WeatherMain">
						<div className="WeatherMainEl">
							<div className="WeatherParam">Weather</div>
							<div className="WeatherValue">{WeatherData.Weather}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Humidity (%)</div>
							<div className="WeatherValue">{WeatherData.Humidity}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Pressure (hPa)</div>
							<div className="WeatherValue">{WeatherData.Pressure}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Wind speed (meter/sec)</div>
							<div className="WeatherValue">{WeatherData.Wind}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Longitude </div>
							<div className="WeatherValue">{WeatherData.Longitude}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Latitude</div>
							<div className="WeatherValue">{WeatherData.Latitude}</div>
						</div>
					</div>
				</div>
			);
		}
		else {
			return (
				<div className="WeatherData">
					<div className="WeatherTop">
						<h2>{WeatherData.CityName}</h2>
						<img src={"http://openweathermap.org/img/wn/" + WeatherData.Icon + ".png"} alt=""></img>
						<h1>{WeatherData.Temperature} &#8451;</h1>
						<button type="submit" className="btn btn-secondary" onClick={() => {
							console.log("REMOVE");
							//store.dispatch({ type: "FAVORITE_REMOVE", payload: WeatherData.CityName });
							store.dispatch(ServerDeleteCity(WeatherData.CityName));
						}} >&#10006;</button>
					</div>
					<div className="WeatherMain">
						<div className="WeatherMainEl">
							<div className="WeatherParam">Weather</div>
							<div className="WeatherValue">{WeatherData.Weather}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Humidity (%)</div>
							<div className="WeatherValue">{WeatherData.Humidity}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Pressure (hPa)</div>
							<div className="WeatherValue">{WeatherData.Pressure}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Wind speed (meter/sec)</div>
							<div className="WeatherValue">{WeatherData.Wind}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Longitude </div>
							<div className="WeatherValue">{WeatherData.Longitude}</div>
						</div>
						<div className="WeatherMainEl">
							<div className="WeatherParam">Latitude</div>
							<div className="WeatherValue">{WeatherData.Latitude}</div>
						</div>
					</div>
				</div>
			);
		}

	}
}