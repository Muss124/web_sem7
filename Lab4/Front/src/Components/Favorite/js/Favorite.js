import React from 'react';
import { FavoriteSearch } from "../../FavoriteSearch/js/FavoriteSearch"
import { connect } from 'react-redux'
import { FavoriteItem } from "../../FavoriteItem/js/FavoriteItem"
import '../css/Favorite.css'
import FetchFavorite from "../../../Actions/FetchFavorite"
import store from "../../../store"

class Favorite extends React.Component {
    constructor(props) {
        super(props);
        store.dispatch(FetchFavorite());
    }
    render() {
        return (
            <div>
                <FavoriteSearch />
                <div className="FavoriteContainer" >
                    {
                        this.props.data.map((favCity, i = 0) => {
                            i++;
                            return ([

                                <FavoriteItem value={favCity} key={i} />

                            ])
                        })
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        data: state
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        setName: () => {
            dispatch({
            });
        }
    };
}
export default connect(mapStateToProps, mapDispatchToProps)(Favorite);