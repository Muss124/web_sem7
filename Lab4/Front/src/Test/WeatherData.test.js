import React from 'react';
import { WeatherData } from '../Components/WeatherData/js/WeatherData';
import renderer from 'react-test-renderer';

describe('Testing render WeatherData', function () {
  it("Header weather", () => {
    const weatherData = { CityName: "Kiev", Temperature: 2.98, Weather: "light intensity drizzle", Humidity: 93, Pressure: 1014, Wind: 4, Longitude: 30.52, Latitude: 50.43, Icon: "09n", Cod: 200 };
    const tree = renderer.create(<WeatherData value={weatherData} isMain={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Favorite weather", () => {
    const weatherData = { CityName: "Kiev", Temperature: 2.98, Weather: "light intensity drizzle", Humidity: 93, Pressure: 1014, Wind: 4, Longitude: 30.52, Latitude: 50.43, Icon: "09n", Cod: 200 };
    const tree = renderer.create(<WeatherData value={weatherData} isMain={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});