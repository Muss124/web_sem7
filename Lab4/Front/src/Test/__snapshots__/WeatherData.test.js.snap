// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render WeatherData Correct input 1`] = `
<div
  class="WeatherData"
>
  <div
    class="WeatherTop"
  >
    <h2>
      Moscow
    </h2>
    <img
      alt=""
      src="http://openweathermap.org/img/wn/10n.png"
    />
    <h1>
      1.45
       ℃
    </h1>
    <button
      class="btn btn-secondary"
      onClick={[Function]}
      type="submit"
    >
      ✖
    </button>
  </div>
  <div
    class="WeatherMain"
  >
    <div
      class="WeatherMainEl"
    >
      <div
        class="WeatherParam"
      >
        Weather
      </div>
      <div
        class="WeatherValue"
      >
        heavy intensity rain
      </div>
    </div>
    <div
      class="WeatherMainEl"
    >
      <div
        class="WeatherParam"
      >
        Humidity (%)
      </div>
      <div
        class="WeatherValue"
      >
        100
      </div>
    </div>
    <div
      class="WeatherMainEl"
    >
      <div
        class="WeatherParam"
      >
        Pressure (hPa)
      </div>
      <div
        class="WeatherValue"
      >
        1012
      </div>
    </div>
    <div
      class="WeatherMainEl"
    >
      <div
        class="WeatherParam"
      >
        Wind speed (meter/sec)
      </div>
      <div
        class="WeatherValue"
      >
        3
      </div>
    </div>
    <div
      class="WeatherMainEl"
    >
      <div
        class="WeatherParam"
      >
        Longitude 
      </div>
      <div
        class="WeatherValue"
      >
        37.62
      </div>
    </div>
    <div
      class="WeatherMainEl"
    >
      <div
        class="WeatherParam"
      >
        Latitude
      </div>
      <div
        class="WeatherValue"
      >
        55.75
      </div>
    </div>
  </div>
</div>
`;
